rfcs/proposed/parallel_block_for_task_arena/README.md

```diff
@@ -62,100 +62,111 @@ aspect of oneTBB behavior should be able to do it.
+    * If work was submitted immediately after the end of the parallel phase,
+      the default arena behavior with regard to "workers leave" policy is restored.
+    * If the default "workers leave" policy was the "Fast leave", the result is NOP.
 
 
 ### Proposed API
 
+Summary of API changes:
+
+* Add enumeration class for the arena leave policy.
+* Add the policy as the last parameter to the arena constructor and initializer
+defaulted to "automatic".
+* Add functions to start and end the parallel phase to the `task_arena` class
+and the `this_task_arena` namespace.
+* Add RAII class to map a parallel phase to a code scope.
+
 ```cpp
 class task_arena {
-    enum class workers_leave : /* unspecified type */ {
+    enum class leave_policy : /* unspecified type */ {
+        automatic = /* unspecifed */,
         fast = /* unspecifed */,
         delayed = /* unspecifed */
     };
 
     task_arena(int max_concurrency = automatic, unsigned reserved_for_masters = 1,
                priority a_priority = priority::normal,
-               workers_leave a_workers_leave = workers_leave::delayed);
+               leave_policy a_leave_policy = leave_policy::automatic);
 
     task_arena(const constraints& constraints_, unsigned reserved_for_masters = 1,
                priority a_priority = priority::normal,
-               workers_leave a_workers_leave = workers_leave::delayed);
+               leave_policy a_leave_policy = leave_policy::automatic);
 
     void initialize(int max_concurrency, unsigned reserved_for_masters = 1,
                     priority a_priority = priority::normal,
-                    workers_leave a_workers_leave = workers_leave::delayed);
+                    leave_policy a_leave_policy = leave_policy::automatic);
 
     void initialize(constraints a_constraints, unsigned reserved_for_masters = 1,
                     priority a_priority = priority::normal,
-                    workers_leave a_workers_leave = workers_leave::delayed);
+                    leave_policy a_leave_policy = leave_policy::automatic);
 
-    void start_parallel_block();
-    void end_parallel_block(bool set_one_time_fast_leave = false);
+    void start_parallel_phase();
+    void end_parallel_phase(bool with_fast_leave = false);
 
-    class scoped_parallel_block {
-        scoped_parallel_block(task_arena& ta, bool set_one_time_fast_leave = false);
+    class scoped_parallel_phase {
+        scoped_parallel_phase(task_arena& ta, bool with_fast_leave = false);
     };
 };
 
 namespace this_task_arena {
-    void start_parallel_block();
-    void end_parallel_block(bool set_one_time_fast_leave = false);
+    void start_parallel_phase();
+    void end_parallel_phase(bool with_fast_leave = false);
 }
 ```
 
-By the contract, users should indicate the end of _parallel block_ for each
-previous start of _parallel block_.<br>
+By the contract, users should indicate the end of _parallel phase_ for each
+previous start of _parallel phase_.<br>
```

Comment:
Maybe "The parallel phase continues until for each previous `start_parallel_phase` call a matching `end_parallel_phase` call has been made."

Reply:
What about "The _parallel phase_ continues until each previous `start_parallel_phase` call has a matching `end_parallel_phase` call"?